fennel/client_tests/test_incremental.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+
+import pytest
+
+from fennel.connectors import Webhook, source
+from fennel.datasets import dataset, field
+from fennel.testing import mock
+
+__owner__ = "nitin@fennel.ai"
+webhook = Webhook(name="fennel_webhook")
+
+
+@pytest.mark.integration
+@mock
+def test_dataset_version_change(client):
+
+    @source(
+        webhook.endpoint("Dataset1"),
+        disorder="14d",
+        cdc="append",
+    )
+    @dataset
+    class Dataset1:
+        user_id: int
+        email: str
+        created_at: datetime
+
+    client.commit(datasets=[Dataset1], message="first_commit")
+
+    # Try syncing the same dataset through incremental should work.
+    client.commit(
+        datasets=[Dataset1], incremental=True, message="second_commit"
+    )
+
+    # Try changing the dataset without increasing the version should fail
+    @source(
+        webhook.endpoint("Dataset1"),
+        disorder="14d",
+        cdc="upsert",
+    )
+    @dataset
+    class Dataset1:
+        user_id: int = field(key=True)
+        email: str
+        created_at: datetime
+
+    with pytest.raises(ValueError) as e:
+        client.commit(
+            datasets=[Dataset1], incremental=True, message="second_commit"
+        )
+    assert str(e.value) == "Please update version of Dataset Dataset1."
+
+    # Try changing the dataset with increasing the version should pass
+    @source(
+        webhook.endpoint("Dataset1"),
+        disorder="14d",
+        cdc="upsert",
+    )
+    @dataset(version=2)
+    class Dataset1:
+        user_id: int = field(key=True)
+        email: str
```

Comment:
can you also add a test where we incrementally change an extractor logic, to test out the read side

Reply:
added